### ISSUE TITLE
[java] CommentSize triggers "Too many lines" on method with several parameters

### ISSUE THREAD

**Issue by mcandre** [pmd/pmd#1607]
**Affects PMD Version:** 

5+

**Rule:**

CommentSize

**Description:**

When the user is documenting a method which takes several parameters, noting `@param`... line by line, then PMD complains that the comment has "Too many lines". Now, there is something to be said for a method with too high an arity, but that should be treated as a separate kind of warning. Hilariously enough, this warning can be silenced by omitting the `@param` documentation for the method! Though of course that fails Checkstyle (should that not fail PMD?)

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Gradle

**Comment by jsotuyod** [comment]
This one is interesting… I agree the arity of the method should be checked by a separate rule in any case, and not be punished by `CommentSize`. I wonder however how the doc for those params should be accounted for…

```java
/**
 * @param foo The foo
 * @param bar The bar
 * @param baz The baz
 * …
*/
```

is clearly ok, but doing:

```java
/**
 * @param foo Lorem ipsum dolor sit amet, consectetur adipiscing elit.
 *            Suspendisse ac molestie leo. Curabitur auctor, nisi ac tempor ultricies,
 *            lacus nisl pulvinar libero, id vulputate quam libero ac ex. Aliquam eu
 *            augue sed urna pharetra faucibus et in sapien. Curabitur rutrum sapien
 *            leo, vehicula fermentum neque hendrerit quis. Vestibulum in tempus
 *            sapien, ac ullamcorper nunc. Suspendisse commodo suscipit scelerisque.
 *            Ut maximus erat vel sapien aliquet eleifend. Donec gravida id arcu ac
 *            pharetra. Morbi porta metus eget ligula blandit, non aliquam ligula blandit.
 *            …
*/
```

probably isn't.

Maybe we should add new properties to allow and set individually limits for each `@param`, `@return`, `@throws`, etc.? possible at 2 lines for those, 1 for `@see` and keeping the current property for the description?

**Comment by bohni** [comment]
My suggestion is to count only lines that don't have a Javadoc tag.
